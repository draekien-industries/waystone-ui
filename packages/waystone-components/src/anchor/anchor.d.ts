/** @jsxImportSource theme-ui */
import { FC } from 'react';
import { LinkProps } from 'theme-ui';
/** The props for the anchor component. */
export interface AnchorProps extends LinkProps {
    /**
     * A flag indicating whether the href is an external link.
     * @default false
     */
    external?: boolean;
}
export declare const Anchor: FC<AnchorProps>;
//# sourceMappingURL=anchor.d.ts.map