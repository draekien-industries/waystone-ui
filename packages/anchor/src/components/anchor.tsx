import { AbsoluteAnchor } from './anchor.absolute';
import { RelativeAnchor } from './anchor.relative';

export const Anchor = {
  Relative: RelativeAnchor,
  Absolute: AbsoluteAnchor,
} as const;
