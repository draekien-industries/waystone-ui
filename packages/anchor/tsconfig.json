{
  "extends": "@waystone/tsconfig/react-library.json",
  "include": ["src", "./jest.setup.ts"],
  "exclude": [
    "dist",
    "build",
    "node_modules",
    "storybook-static",
    ".storybook",
    "__snapshots__",
    "__assets__",
    "__tests__",
    "__fixtures__",
    "**/*.stories.tsx",
    "**/*.{spec,test,internal}.tsx"
  ],
  "compilerOptions": {
    "outDir": "dist",
  }
}
