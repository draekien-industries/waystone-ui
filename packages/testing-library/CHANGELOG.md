# @waystone/testing-library

## 9.2.0

### Minor Changes

- 21db598: chore: bump deps and add documentation

### Patch Changes

- Updated dependencies [21db598]
  - @waystone/theme@9.3.0

## 9.1.6

### Patch Changes

- 41664e2: fix: use type imports to improve bundling
- Updated dependencies [41664e2]
  - @waystone/theme@9.2.5

## 9.1.5

### Patch Changes

- 856ba2b: fix: missing dependencies when installing package
- Updated dependencies [856ba2b]
  - @waystone/theme@9.2.4

## 9.1.4

### Patch Changes

- 5b726aa: fix build and dependencies
- Updated dependencies [5b726aa]
  - @waystone/theme@9.2.3

## 9.1.3

### Patch Changes

- 7086019: Update dependencies
- Updated dependencies [7086019]
  - @waystone/theme@9.2.2

## 9.1.2

### Patch Changes

- @waystone/theme@9.2.1

## 9.1.1

### Patch Changes

- Updated dependencies [8f75dd6]
  - @waystone/theme@9.2.0

## 9.1.0

### Minor Changes

- 5c43357: Switch to pnpm and implement minimal syntax highlighter

### Patch Changes

- Updated dependencies [5c43357]
  - @waystone/theme@9.1.0

## 9.0.0

### Major Changes

- 4a19c42: Update package structure

  Created the `@waystone/ui` package for easier installation and created the `@waystone/testing-library`
  package for a preconfigured testing-library render function.

  Uninstall your `@waystone/*` packages, and run the command `yarn add @waystone/ui`
  to update to the latest version. You will need to update the imports to point
  to the new source.

### Patch Changes

- Updated dependencies [4a19c42]
  - @waystone/theme@9.0.0
