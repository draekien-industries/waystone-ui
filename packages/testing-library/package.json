{
  "name": "@waystone/testing-library",
  "version": "10.0.2",
  "private": true,
  "main": "./src/index.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/draekien-industries/waystone-ui"
  },
  "description": "The react component library used by draekien-industries",
  "author": "draekien-industries",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "src/**"
  ],
  "scripts": {
    "lint": "eslint --ext .tsx,.ts,.jsx,.js ./src/ --fix",
    "lint:release": "eslint --ext .tsx,.ts,.jsx,.js ./src/"
  },
  "devDependencies": {
    "@waystone/tsconfig": "workspace:*",
    "eslint-config-waystone": "workspace:*",
    "eslint-plugin-waystone": "workspace:*",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "dependencies": {
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.2.0",
    "@testing-library/user-event": "^14.6.0",
    "@theme-ui/color": "^0.17.1",
    "@theme-ui/css": "^0.17.1",
    "@waystone/theme-provider": "workspace:*",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "theme-ui": "^0.17.1"
  },
  "publishConfig": {
    "access": "public"
  }
}
