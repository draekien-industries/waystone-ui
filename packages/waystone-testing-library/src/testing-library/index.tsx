'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
