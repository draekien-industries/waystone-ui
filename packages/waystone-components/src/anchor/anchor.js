var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { jsx as _jsx } from "theme-ui/jsx-runtime";
import { Link } from 'theme-ui';
import { anchorCss } from './anchor.styles';
/** The internal anchor component */
const InternalAnchor = (_a) => {
    var { children } = _a, rest = __rest(_a, ["children"]);
    return (_jsx(Link, Object.assign({}, rest, { sx: anchorCss }, { children: children })));
};
/** The external anchor component */
const ExternalAnchor = (_a) => {
    var { children, target = '_blank', rel = 'noopener noreferrer' } = _a, rest = __rest(_a, ["children", "target", "rel"]);
    return (_jsx(Link, Object.assign({}, rest, { sx: anchorCss, target: target, rel: rel }, { children: children })));
};
export const Anchor = (_a) => {
    var { external = false } = _a, rest = __rest(_a, ["external"]);
    if (external) {
        return _jsx(ExternalAnchor, Object.assign({}, rest));
    }
    return _jsx(InternalAnchor, Object.assign({}, rest));
};
