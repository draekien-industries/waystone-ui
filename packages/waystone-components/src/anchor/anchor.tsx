/** @jsxImportSource theme-ui */
import { HTMLAttributeAnchorTarget, ReactNode } from 'react';
import { Link, LinkProps } from 'theme-ui';
import { Icon } from '../icon/icon';
import { anchorCss, undecoratedAnchorCss } from './anchor.styles';

/**
 * The props for the anchor component.
 * @extends LinkProps
 */
export interface AnchorProps extends LinkProps {
  /**
   * A flag indicating whether the href is an external link.
   * @default false
   */
  external?: boolean;
  /**
   * The target to open the link in. The default value only applies when the `external` flag is `true`.
   */
  target?: HTMLAttributeAnchorTarget;
  /**
   * The relationship between the linked resource and the current document.
   * The default value only applies when the `external` flag is `true`.
   */
  rel?: string;
  /**
   * The react component to display inside the anchor.
   */
  children: ReactNode;
  /** Should the anchor render without css styles */
  noDecoration?: boolean;
}

/** The internal anchor component */
const InternalAnchor = ({
  children,
  noDecoration,
  ...rest
}: LinkProps & Pick<AnchorProps, 'noDecoration'>) => (
  <Link {...rest} sx={noDecoration ? undecoratedAnchorCss : anchorCss}>
    {children}
  </Link>
);

/** The external anchor component */
const ExternalAnchor = ({
  children,
  noDecoration,
  target = '_blank',
  rel = 'noopener noreferrer',
  ...rest
}: LinkProps & Pick<AnchorProps, 'noDecoration'>) => (
  <Link
    {...rest}
    sx={!noDecoration ? anchorCss : undecoratedAnchorCss}
    target={target}
    rel={rel}
  >
    {children} <Icon name="open_in_new" size="sm" sx={{ pb: 'xs' }} />
  </Link>
);

/**
 * A styled HTML `<a>` element which creates a hyperlink to the provided URL.
 * Content passed in via the `children` prop should indicate the link's destination.
 * See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a for more details
 * about the HTML `<a>` element.
 */
export const Anchor = ({ external = false, ...rest }: AnchorProps) => {
  if (external) {
    return <ExternalAnchor {...rest} />;
  }

  return <InternalAnchor {...rest} />;
};
