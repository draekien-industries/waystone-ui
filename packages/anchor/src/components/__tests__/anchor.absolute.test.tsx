import { render, screen } from '@waystone/testing-library';
import { Anchor } from '../anchor';

describe('<Anchor.Absolute />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Anchor.Absolute
        href="https://example.com"
        referrerPolicy="no-referrer"
        target="">
        link
      </Anchor.Absolute>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with no decoration', () => {
    const { container } = render(
      <Anchor.Absolute
        href="https://example.com"
        referrerPolicy="no-referrer"
        target=""
        nodecoration>
        link
      </Anchor.Absolute>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the external link icon', () => {
    render(
      <Anchor.Absolute
        href="https://example.com"
        referrerPolicy="no-referrer"
        target=""
        nodecoration>
        link
      </Anchor.Absolute>
    );

    const icon = screen.getByText('open_in_new');

    expect(icon).toBeInTheDocument();
  });
});
