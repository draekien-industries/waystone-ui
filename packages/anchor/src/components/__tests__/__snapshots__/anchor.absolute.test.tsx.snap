// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor.Absolute /> should match snapshot 1`] = `
<div>
  <a
    class="css-158cs26-AbsoluteAnchorWithRef"
    href="https://example.com"
    referrerpolicy="no-referrer"
    target=""
  >
    <i
      class="material-icons-outlined css-1s21fbt-AbsoluteAnchorWithRef"
    >
      open_in_new
    </i>
    link
  </a>
</div>
`;

exports[`<Anchor.Absolute /> should render with no decoration 1`] = `
<div>
  <a
    class="css-18kxuu6-AbsoluteAnchorWithRef"
    href="https://example.com"
    referrerpolicy="no-referrer"
    target=""
  >
    <i
      class="material-icons-outlined css-1s21fbt-AbsoluteAnchorWithRef"
    >
      open_in_new
    </i>
    link
  </a>
</div>
`;
